tests/functional/n/none_dunder_protocols_named_expr.py

```diff
@@ -0,0 +1,13 @@
+"""Although unlikely to happen, this test case was almost missed in development."""
```

Comment:
I'm torn on whether this test is truly worth it 🤔

Reply:
I think if it prevents a crash from going undetected it's worth it 😄 Some codebases have truly horrific code but pylint need to be all terrains. Our typing is not yet capable of warning of everything but I think we should not silence it with assert, even if admittedly this kind of code construct won't happen very often in the wild, and won't happen with sane lib.